PeakCoreData/Core/Cache/Cache.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  Cache.swift
+//  PeakCoreData-iOS
+//
+//  Created by David Yates on 02/04/2020.
+//  Copyright © 2020 3Squared Ltd. All rights reserved.
+//
+
+// Adapted from: https://www.swiftbysundell.com/articles/caching-in-swift/
+
+import Foundation
+
+/// Provides a wrapped around `NSCache` that doesn't require keys to subclass `NSObject`.
+public final class Cache<Key: Hashable, Value> {
+    
+    private let wrapped = NSCache<WrappedKey, Entry>()
```

Comment:
Is there any kind of memory limit or eviction policy that should be set on NSCache, or that a use should be able to set?

Reply:
I've added properties for `name`, `countLimit` and `totalCostLimit` as well as an optional block that will be called as objects are removed.